src/software/simulation/er_force_simulator.cpp

```diff
@@ -298,13 +300,67 @@ void ErForceSimulator::stepSimulation(const Duration& time_step)
     SSLSimulationProto::RobotControl blue_robot_control =
         updateSimulatorRobots(blue_simulator_robots, *blue_team_vision_msg);
 
-    er_force_sim->acceptYellowRobotControlCommand(yellow_robot_control);
-    er_force_sim->acceptBlueRobotControlCommand(blue_robot_control);
+    auto yellow_radio_responses =
+        er_force_sim->acceptYellowRobotControlCommand(yellow_robot_control);
+    auto blue_radio_responses =
+        er_force_sim->acceptBlueRobotControlCommand(blue_robot_control);
+
+    blue_robot_with_ball.reset();
+    yellow_robot_with_ball.reset();
+
+    for (const auto& response : yellow_radio_responses)
+    {
+        if (response.has_ball_detected() && response.ball_detected())
+        {
+            yellow_robot_with_ball = response.id();
+        }
+    }
+
+    for (const auto& response : blue_radio_responses)
+    {
+        if (response.has_ball_detected() && response.ball_detected())
+        {
+            blue_robot_with_ball = response.id();
+        }
+    }
+
     er_force_sim->stepSimulation(time_step.toSeconds());
 
     frame_number++;
 }
 
+std::vector<TbotsProto::RobotStatus> ErForceSimulator::getBlueRobotStatuses() const
+{
+    std::vector<TbotsProto::RobotStatus> robot_statuses;
```

Comment:
Curious why this is a `vector` if only 1 element is added to it? Same with `ErForceSimulator::getYellowRobotStatuses`

Reply:
it's possible that we populate more robot statuses in the future with not just breakbeam, so this is an abstraction with not much overhead